patchwork/views/patch.py

```diff
@@ -136,6 +136,13 @@ def patch_detail(request, project_id, msgid):
         related_same_project = []
         related_different_project = []
 
+    context['next_submission'] = None
+    patch_series = patch.series.patches.all()
+    num_patches = patch.series.patches.count()
+    for idx in range(num_patches - 1):
```

Comment:
To the best of my knowledge, the sort order for this is based on `Patch.Meta.ordering`, which is `date`. `Patch.date` is typically taken from the send date of the email, not the received date. I think it's fair to say patch 1/N will always have an earlier/the same `date` as patch 2/N, as so on. Might be worth double checking though.

(fwiw, I think this is also an issue for the `Series` pane)